### ISSUE TITLE
bug: ion-back-button ignores global config for backButtonText

### ISSUE THREAD

**Issue by vadimwe** [ionic-team/ionic-framework#24052]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already report this problem, without success.

### Ionic Framework Version

- [ ] v4.x
- [X] v5.x
- [ ] v6.x

### Current Behavior

ion-back-button ignores global config:

`IonicModule.forRoot({
      backButtonText: '',
    }),`

### Expected Behavior

back button contains only icon

### Steps to Reproduce

Update to "@ionic/angular": "^5.8.3" with "@angular/*": "~12.2.9"

### Ionic Info

Ionic:

   Ionic CLI                     : 6.17.1 (/usr/local/lib/node_modules/@ionic/cli)
   Ionic Framework               : @ionic/angular 5.8.3
   @angular-devkit/build-angular : 12.2.9
   @angular-devkit/schematics    : 12.0.3
   @angular/cli                  : 12.2.9
   @ionic/angular-toolkit        : 4.0.0

Cordova:

   Cordova CLI       : 10.0.0
   Cordova Platforms : android 10.1.0, ios 6.2.0
   Cordova Plugins   : cordova-plugin-ionic-keyboard 2.2.0, cordova-plugin-ionic-webview 5.0.0, (and 16 other plugins)

Utility:

   cordova-res : 0.15.3
   native-run  : 1.4.1

System:

   Android SDK Tools : 26.1.1 (/Users/vs/Library/Android/sdk)
   ios-deploy        : 1.11.4
   ios-sim           : 8.0.2
   NodeJS            : v12.18.3 (/usr/local/bin/node)
   npm               : 6.14.6
   OS                : macOS Big Sur
   Xcode             : Xcode 13.0 Build version 13A233

**Comment by chenwei116057** [comment]
I also encountered this problem

**Comment by vadimwe** [comment]
Quick fix if you need.

variables.scss or another global file:
`ion-back-button {
  &::part(text) {
    display: none;
  }
}`

**Comment by EinfachHans** [comment]
I debugged a bit. Looks like [this commit](https://github.com/ionic-team/ionic-framework/commit/f112ad4490dc4a179dc3feab495530e14e655e5a) causes this.

`config.reset(configObj);` in the **initialize** is first called with: 

```
_zoneGate: function(),
backButtonText: "",
persistConfig: false
```

and then called again (from `defineCustomElements.globalScripts`) with:

```
m: Map {"persistConfig" => false, "backButtonText" => "", "_zoneGate" => function, "mode" => "ios"} ,
persistConfig: false
```

**Comment by liamdebeasi** [comment]
Thanks for the issue. This has been resolved via https://github.com/ionic-team/ionic-framework/pull/24054, and a fix will be available in an upcoming release of Ionic Framework.

**Comment by liamdebeasi** [comment]
This is fixed in Ionic Angular v5.8.4. Thanks for the report!

```
npm install @ionic/angular@latest
```
